search-api/search-core/src/main/java/org/sunbird/search/processor/SearchProcessor.java

```diff
@@ -328,6 +436,26 @@ private QueryBuilder prepareSearchQuery(SearchDTO searchDTO) {
 		return boolQuery;
 	}
 
+	private QueryBuilder prepareSearchQuery(SearchDTO searchDTO, boolean fuzzyEnabled) {
+		BoolQueryBuilder boolQuery = new BoolQueryBuilder();
+		QueryBuilder queryBuilder = null;
+		String totalOperation = searchDTO.getOperation();
+		List<Map> properties = searchDTO.getProperties();
+		formQuery(properties, queryBuilder, boolQuery, totalOperation, fuzzyEnabled);
+		if (searchDTO.getMultiFilterProperties() != null) {
+			formQuery(searchDTO.getMultiFilterProperties(), queryBuilder, boolQuery,
+					SearchConstants.SEARCH_OPERATION_OR, fuzzyEnabled);
```

Comment:
In the existing "prepareSearchQuery" method, the last param for "formQuery" method call uses searchDTO.isFuzzySearch() -- which is the proper way of coding... by introducing the "fuzzyEnabled" - which is hardcoded the existing flow is not using "fuzzy" by default and "fuzzy" automatically used if search results are zero.